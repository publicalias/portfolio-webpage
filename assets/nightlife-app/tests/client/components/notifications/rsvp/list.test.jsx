"use strict";

//local imports

const List = require("../../../../../scripts/client/components/notifications/rsvp/list");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { testMock } = require("redux/tests/meta-tests");
const { initTestEvent, initTestSnapshot, reactTests } = require("redux/tests/react-tests");

//utilities

const { testMount, testShallow } = testWrapper(List);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(List));

//list

describe("list", () => {

  const testSnapshot = initTestSnapshot(testShallow);

  const testClick = initTestEvent(testMount, "click");

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (rsvps)", () => testSnapshot({ notifications: { rsvps: [{}] } }));

  it("should call rsvpGetList on load", () => {

    const { props, wrapper } = testMount();

    const { actions: { rsvpGetList } } = props;

    wrapper.mount();

    testMock(rsvpGetList, []);

    wrapper.unmount();

  });

  it("should call rsvpGetList on click", () => testClick(".qa-sync-rsvp", [], [
    "rsvpGetList",
    [], //on load
    []
  ]));

});
