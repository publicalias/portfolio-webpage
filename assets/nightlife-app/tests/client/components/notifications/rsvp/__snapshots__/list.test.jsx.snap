// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should match snapshot (default) 1`] = `
<div
  className="c-rsvp-list"
>
  <div
    className="c-rsvp-list__head"
  >
    <h3>
      RSVPs
    </h3>
    <button
      className="c-rsvp-list__sync qa-sync-rsvp"
      onClick={[Function]}
    >
      <i
        className="fas fa-sync"
      />
    </button>
  </div>
  <hr />
  <div
    className="c-rsvp-list__body"
  >
    <p
      className="u-margin-none"
    >
      No notifications
    </p>
  </div>
</div>
`;

exports[`list should match snapshot (rsvps) 1`] = `
<div
  className="c-rsvp-list"
>
  <div
    className="c-rsvp-list__head"
  >
    <h3>
      RSVPs
    </h3>
    <button
      className="c-rsvp-list__sync qa-sync-rsvp"
      onClick={[Function]}
    >
      <i
        className="fas fa-sync"
      />
    </button>
  </div>
  <hr />
  <div
    className="c-rsvp-list__body"
  >
    <Item
      actions={
        Object {
          "favoriteAdd": [MockFunction],
          "favoriteRemove": [MockFunction],
          "friendAdd": [MockFunction],
          "friendConfirm": [MockFunction],
          "friendDismiss": [MockFunction],
          "friendGetList": [MockFunction],
          "friendRemove": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSaveAddress": [MockFunction],
          "metaSaveAvatar": [MockFunction],
          "metaSetAddress": [MockFunction],
          "metaSetAvatar": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "metaToggleDelete": [MockFunction],
          "metaToggleSettings": [MockFunction],
          "rsvpAdd": [MockFunction],
          "rsvpDismiss": [MockFunction],
          "rsvpGetList": [MockFunction],
          "rsvpRemove": [MockFunction],
          "userClearState": [MockFunction],
          "userGetItem": [MockFunction],
          "userGetList": [MockFunction],
          "userSetSearch": [MockFunction],
          "userToggleBlock": [MockFunction],
          "venueClearState": [MockFunction],
          "venueGetItem": [MockFunction],
          "venueGetList": [MockFunction],
          "venueSetMessage": [MockFunction],
          "venueSetSearch": [MockFunction],
          "venueSetTime": [MockFunction],
          "venueToggleForm": [MockFunction],
        }
      }
      data={
        Object {
          "account": Object {
            "address": "",
            "avatar": "",
            "delete": false,
            "settings": false,
          },
          "data": Object {
            "users": Array [],
            "venues": Array [],
          },
          "errors": Array [],
          "loading": false,
          "notifications": Object {
            "friends": Array [],
            "rsvps": Array [
              Object {
                "date": 0,
                "hidden": Array [],
                "id": "",
                "message": "",
                "time": "",
                "user": Object {
                  "id": "",
                  "name": "",
                },
                "venue": Object {
                  "id": "",
                  "name": "",
                },
              },
            ],
          },
          "user": Object {},
          "users": Object {
            "list": Object {
              "search": "",
            },
          },
          "venues": Object {
            "list": Object {
              "search": "",
            },
            "page": Object {
              "form": false,
              "message": "",
              "time": "",
            },
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      key="0-0"
      local={
        Object {
          "rsvp": Object {
            "date": 0,
            "hidden": Array [],
            "id": "",
            "message": "",
            "time": "",
            "user": Object {
              "id": "",
              "name": "",
            },
            "venue": Object {
              "id": "",
              "name": "",
            },
          },
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </div>
</div>
`;
